src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -138,14 +166,14 @@ public void SetPinMode(int pinNumber, PinMode mode)
         /// </summary>
         /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
         /// <returns>The mode of the pin.</returns>
-        public PinMode GetPinMode(int pinNumber)
+        public virtual PinMode GetPinMode(int pinNumber)
         {
-            int logicalPinNumber = GetLogicalPinNumber(pinNumber);
-            if (!_openPins.Contains(logicalPinNumber))
+            if (!IsPinOpen(pinNumber))
             {
-                throw new InvalidOperationException($"Can not get the mode of pin {logicalPinNumber} because it is not open.");
+                throw new InvalidOperationException($"Can not get the mode of pin {pinNumber} because it is not open.");
             }
 
+            int logicalPinNumber = GetLogicalPinNumber(pinNumber);
             return _driver.GetPinMode(logicalPinNumber);
```

Comment:
Maybe all this drive access can be exposed via protected methods like *FromDriver, eg `GetPinModeFromDriver`, `OpenPinFromDriver` or at least give access to the driver instance via a protected property?

Reply:
That adds a lot of complexity to the interface. Why would you need that?

Reply:
Yeah, I agree that those would just add complexity and don't see a good use case for them.